src/main/java/org/jboss/msc/service/management/ServiceStatus.java

```diff
@@ -194,27 +262,15 @@ public String getException() {
      */
     public String toString() {
         final StringBuilder builder = new StringBuilder();
-        builder.append("Service \"").append(serviceName).append("\" ");
-        final String[] aliases = this.aliases;
-        if (aliases.length > 0) {
-            builder.append("(aliases: ");
-            for (int i = 0; i < aliases.length; i++) {
-                builder.append(aliases[i]);
-                if (i < aliases.length - 1) {
-                    builder.append(", ");
-                }
-            }
-            builder.append(") ");
-        }
+        builder.append("Service names: \"").append(Arrays.asList(serviceNames)).append("\" ");
         builder.append("(class ").append(serviceClassName).append(')');
         builder.append(" mode ").append(modeName);
         builder.append(" state ").append(stateName);
         if (! stateName.equals(substateName)) {
             builder.append(" (").append(substateName).append(')');
         }
-        String parentName = this.parentName;
-        if (parentName != null) {
-            builder.append(" (parent: ").append(parentName).append(')');
+        if (parentNames != null) {
+            builder.append(" (parent names: ").append(Arrays.asList(parentNames)).append(')');
```

Comment:
This is another example where having a name for the controller itself might be useful.